giap.client/src/main.tsx

```diff
@@ -0,0 +1,35 @@
+import {StrictMode, useEffect} from 'react'
+import {createRoot} from 'react-dom/client'
+import './index.css'
+import App from './App.tsx'
+import {BrowserRouter, Routes, Route, useParams} from 'react-router';
+import {useTranslation} from 'react-i18next';
+
+// Wrapper that sets the language based on the URL
+// from IBAN issuer: https://github.com/privacybydesign/yivi-iban-issuer/tree/main/react-cra
+function LanguageRouter() {
+    const {lang} = useParams();
+    const {i18n} = useTranslation();
+
+    useEffect(() => {
+        if (lang && i18n.language !== lang) {
+            i18n.changeLanguage(lang);
+        }
+    }, [lang, i18n]);
+
+    return (
+        <Routes>
+            <Route path="/:slug" element={<App/>}/>
+        </Routes>
+    );
+}
+
+createRoot(document.getElementById('root')!).render(
+    <StrictMode>
+        <BrowserRouter>
+            <Routes>
+                <Route path=":lang/*" element={<LanguageRouter/>}/>
```

Comment:
I think you are missing a redirect for a defualt language. What if someone navigates to the root of this website then the user should be rerouted to /en or /nl